.github/workflows/format-code.yml

```diff
@@ -0,0 +1,47 @@
+name: "Format C code"
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  format:
+    name: "Format C code"
+    runs-on: ubuntu-18.04
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/format')
+    steps:
+      - name: "Checkout code"
+        run: |
+          PR_DATA="/tmp/pr.json"
```

Comment:
@patricksjackson I've lifted this from the C++ workflow you linked on #472. Do you know why the checkout action isn't used instead of all this?
I suspect that maybe checkout action does not get PR code, but just the current master, but I need to learn about the action.

Reply:
Had a look at the action (https://github.com/marketplace/actions/checkout) and the `issue_comment` event spec (https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#issue_comment).
Seems my guess above is essentially the case.
Looks like the action may be able to do the same, if I add some additional futzing about, but the difference would then be negligible I think.

Reply:
Makes sense. If there isn't an official way to do it with the checkout action we should probably just stick with this as it already works.